Shader Language: function calls are assignable

**Operating system or device, Godot version, GPU Model and driver (if graphics related):**
Windows 10 - master

**Issue description:**
In the shading language in godot you can assign to a function.
The following shader code will produce no errors:

```
shader_type canvas_item;

void one()
{
}

void two()
{
}

void fragment()
{
	one() = two();
}
```

Yes it does indeed produce a GLSL error in the console. 

```
ERROR: _display_error_with_code: CanvasShaderGLES3: Fragment Program Compilation Failed:
0(245) : error C1034: assignment to non-lvalue
```

--- a/servers/visual/shader_language.cpp
+++ b/servers/visual/shader_language.cpp
@@ -1041,6 +1041,12 @@ bool ShaderLanguage::_validate_operator(OperatorNode *p_op, DataType *r_ret_type
 			}
 		} break;
 		case OP_ASSIGN: {
+
+			if (p_op->arguments[0]->type != Node::TYPE_MEMBER && p_op->arguments[0]->type != Node::TYPE_VARIABLE) {
+				valid = false;
+				break;
+			}
+
 			DataType na = p_op->arguments[0]->get_datatype();
 			DataType nb = p_op->arguments[1]->get_datatype();
 			valid = na == nb;

